### PR TITLE
cleaning up clang-tidy errors

### PR DESCRIPTION
**Pull request by naoyam**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/kernel_ir.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ void Node::print() const {
 }
 
 Val::Val(Passkey passkey, DataType dtype) : Node(passkey), dtype_(dtype) {
-  id_ = passkey.kernel->newValueId(passkey);
+  id_ = passkey.kernel->newValueId(passkey); // NOLINT
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/kernel_ir.cpp]: we can add a link to the clang-tidy bug to explain the need for suppressing the diagnostic:
https://bugs.llvm.org/show_bug.cgi?id=48534

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 2]
```diff
@@ -254,7 +254,10 @@ kir::ExpressionEvaluator bindKernelInputs(
           expr_eval.bind(extent, value);
         }
       }
-    } else if (input->isScalar() && input->dtype() == DataType::Int) {
+    } else if (input->isScalar() && input->dtype() == DataType::Int) { // NOLINT
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp]: did clang-format split the comment over 4 separate lines?

File: torch/csrc/jit/codegen/cuda/arith.cpp [hunk 3]
```diff
@@ -947,7 +947,7 @@ TensorView* sum_to(TensorView* in, const std::vector<Int*>& sum_to_size) {
   bool reduction_within_shape = false;
 
   // Reduce rest of the dims with keep_dim
-  for (int i = leading_dims; i < root.size(); i++) {
+  for (int i = leading_dims; i < (int)root.size(); i++) {
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/arith.cpp]: please use the "functional" syntax (`int(root.size())`) instead, or `static_cast<int>(root.size())` - I'd prefer the former

File: torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp [hunk 4]
```diff
@@ -165,7 +165,7 @@ class AllocateReuseModifier {
           const auto register_size =
               expr_evaluator_.evaluate(allocation->size());
           if (register_size.has_value()) {
-            local_valid = *register_size > kRegisterSizeThreshold;
+            local_valid = (size_t)*register_size > kRegisterSizeThreshold;
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp]: `size_t(*register_size)`

File: torch/csrc/jit/codegen/cuda/parser.cpp [hunk 5]
```diff
@@ -542,18 +542,18 @@ class IrParser {
             }
 
             // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
-            auto training = constant_as<bool>(node->input(5));
-            TORCH_INTERNAL_ASSERT(
-                training.has_value(),
-                "The training (bool) parameter is required.");
-            const bool kTraining = training.value();
+            // auto training = constant_as<bool>(node->input(5));
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/parser.cpp:545]: why keep commented out code? if it's not needed we should delete it (even if we anticipate future uses, we can add it back when we need it)

File: torch/csrc/jit/codegen/cuda/scheduler.cpp [hunk 6]
```diff
@@ -1065,7 +1064,7 @@ void organizeAxes(
   // Move reduction axes to the inner-most position
   merged_reduction_axis = findMergedReductionAxis(first_reduction_tv);
   const size_t kInnerMostAxis = first_reduction_tv->domain()->nDims() - 1;
-  if (merged_reduction_axis != kInnerMostAxis) {
+  if (merged_reduction_axis != (int)kInnerMostAxis) {
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/scheduler.cpp]: `int(...)` or `static_cast<int>(...)`

**Review by tlemo — APPROVED** [review]
Approved with small comments

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 2 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp]: Yes, it seems so.

File: torch/csrc/jit/codegen/cuda/kernel_ir.cpp (hunk 1 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/kernel_ir.cpp]: Changed to use `NOLINTNEXTLINE` as adding the link triggers a new line by clang-format.

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 2 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp]: Changed to use `NOLINTNEXTLINE`.

File: torch/csrc/jit/codegen/cuda/parser.cpp (hunk 5 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/parser.cpp:545]: @rdspring1 wants to keep it there.

**Pull request merged**